sklearn/metrics/_scorer.py

```diff
@@ -298,22 +353,26 @@ def _score(self, method_caller, clf, X, y, sample_weight=None):
             try:
                 y_pred = method_caller(clf, "decision_function", X)
 
-                # For multi-output multi-class estimator
                 if isinstance(y_pred, list):
+                    # For multi-output multi-class estimator
                     y_pred = np.vstack([p for p in y_pred]).T
+                elif y_type == "binary" and "pos_label" in self._kwargs:
+                    self._check_pos_label(
+                        self._kwargs["pos_label"], clf.classes_
+                    )
+                    if self._kwargs["pos_label"] == clf.classes_[0]:
+                        # The implicit positive class of the binary classifier
+                        # does not match `pos_label`: we need to invert the
+                        # predictions
+                        y_pred *= -1
 
             except (NotImplementedError, AttributeError):
                 y_pred = method_caller(clf, "predict_proba", X)
 
                 if y_type == "binary":
-                    if y_pred.shape[1] == 2:
-                        y_pred = y_pred[:, 1]
```

Comment:
Maybe we can simplify this PR by updating this to:

```py
pos_label = self._kwargs.get("pos_label", classes[1])
self._check_pos_label(pos_label, classes)
col_idx = np.flatnonzero(classes == pos_label)[0]
y_pred = y_pred[:, col_idx]
```

in `_ProbaScorer` and `_ThresholdScorer` and do the `_select_proba` refactor in another PR. This way we can get this PR to be smaller while fixing the `pos_label`.

Reply:
The check within ProbaScorer avoid the issue.